modules/tinymce/src/plugins/table/main/ts/ui/CustomColorSwatch.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import Editor from 'tinymce/core/api/Editor';
+import { Menu } from 'tinymce/core/api/ui/Ui';
+
+interface ColorSwatchDialogData {
+  colorpicker: string;
+}
+
+const mapColors = (colorMap: string[]): Menu.ChoiceMenuItemSpec[] => {
```

Comment:
This is duplicated with the Silver logic... we want to avoid duplication if possible, so can this be moved and made a little more generic? The `acid` library would be a good place for that I believe.

Reply:
Added to the external library.